docs/traffic-policy/secrets.mdx

```diff
@@ -0,0 +1,147 @@
+---
+title: Traffic Policy Secrets
+sidebar_label: Secrets
+description: Store sensitive data in encrypted vaults and reference them dynamically in your traffic policies.
+---
+
+Secrets enable you to store sensitive data in encrypted vaults and reference them dynamically in your traffic policies.
+This feature eliminates the need to hardcode passwords, API keys, and other sensitive values directly in policy YAML files.
+
+:::info
+Secrets and vaults are currently in Developer Preview.
+Log into the [ngrok dashboard](https://dashboard.ngrok.com/developer-preview) to request access.
+During the Developer Preview, the following limitations apply:
+
+- Secrets management is only available via the ngrok API (no dashboard UI)
+- Secrets interpolated into certain actions may appear in cleartext in Traffic Inspector when full capture mode is enabled
+- Secrets are only supported in the fields listed in the [supported actions and fields section](#supported-actions-and-fields) below
+  :::
+
+## How it works
+
+Vaults are secure containers that store your secrets.
+Each vault can contain multiple secrets, which are encrypted key-value pairs.
+Secrets are stored using AES-256 encryption at rest and transmitted over HTTPS with TLS 1.2+.
+
+When you reference a secret in a Traffic Policy, the value is evaluated at runtime and never persisted in policy documents or logs.
+The ngrok API never returns secret values in response payloads.
+
+Reference secrets in your Traffic Policy using the `secrets.get()` macro:
+
+```yaml
+secrets.get("vault-name", "secret-name")
+```
+
+The macro dynamically retrieves the secret value from the specified vault at runtime.
+
+## Using vaults and secrets
+
+To use secrets, you'll need to create a vault to store them in.
+
+### Create a vault
+
+Use the ngrok Agent CLI to create a vault:
+
+```shell
+ngrok api vaults create --name "my-vault" --description "Description of vault purpose"
+```
+
+The response includes the vault ID, which you'll need when creating secrets within the vault:
+
+```json
+{
+  "id": "vault_2yNPzuk6GjHrx3mlOCkJK42RsdR",
+  "name": "my-vault",
+  "description": "Description of vault purpose",
+  "created_at": "2025-08-01T21:10:37Z",
+  "uri": "https://api.ngrok.com/vaults/vault_2yNPzuk6GjHrx3mlOCkJK42RsdR"
+}
+```
+
+### Create a secret
+
+Create a secret within a vault using the vault ID:
+
+```shell
+ngrok api secrets create \
+  --name "my-secret" \
+  --value "secret-value" \
+  --vault-id "vault_2yNPzuk6GjHrx3mlOCkJK42RsdR" \
+  --description "Optional description"
+```
+
+### Supported actions and fields
+
+Secrets are supported in the following Traffic Policy actions and fields:
+
+- **basic-auth**: `credentials`
```

Comment:
Should we hyperlink over to each of these actions' documentation from here, you think?